src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -0,0 +1,425 @@
+                    return !dostatement.CloseParenToken.IsMissing;
+
+                default:
+                    // Statement I'm not handling yet so shouldn't proceed with statement completion
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Determines if a syntax node includes all required closing delimiters.
+        /// </summary>
+        /// <remarks>
+        /// <para>Some syntax nodes, such as parenthesized expressions, require a matching closing delimiter to end the
+        /// syntax node. If this node is omitted from the source code, the parser will automatically insert a zero-width
+        /// "missing" closing delimiter token to produce a valid syntax tree. This method determines if required closing
+        /// delimiters are present in the original source.</para>
+        /// </remarks>
+        /// <param name="currentNode"></param>
+        /// <param name="lastDelimiterSpan"></param>
+        /// <returns>
+        /// <list type="bullet">
+        /// <item><description><see langword="true"/> if <paramref name="currentNode"/> requires a closing delimiter and the closing delimiter is present in the source (i.e. not missing)</description></item>
+        /// <item><description><see langword="true"/> if <paramref name="currentNode"/> does not require a closing delimiter</description></item>
+        /// <item><description>otherwise, <see langword="false"/>.</description></item>
+        /// </list>
+        /// </returns>
+        private static bool ClosingDelimiterExistsIfNeeded(SyntaxNode currentNode, ref TextSpan lastDelimiterSpan)
+        {
+            switch (currentNode.Kind())
+            {
+                case SyntaxKind.ArgumentList:
+                    var argumentList = (ArgumentListSyntax)currentNode;
+                    lastDelimiterSpan = argumentList.CloseParenToken.Span;
+                    return !argumentList.CloseParenToken.IsMissing;
+
+                case SyntaxKind.ParenthesizedExpression:
+                    var parenthesizedExpression = (ParenthesizedExpressionSyntax)currentNode;
+                    lastDelimiterSpan = parenthesizedExpression.CloseParenToken.Span;
+                    return !parenthesizedExpression.CloseParenToken.IsMissing;
+
+                case SyntaxKind.BracketedArgumentList:
+                    var bracketedArgumentList = (BracketedArgumentListSyntax)currentNode;
+                    lastDelimiterSpan = bracketedArgumentList.CloseBracketToken.Span;
+                    return !bracketedArgumentList.CloseBracketToken.IsMissing;
+
+                case SyntaxKind.ObjectInitializerExpression:
+                    var initializerExpressionSyntax = (InitializerExpressionSyntax)currentNode;
+                    lastDelimiterSpan = initializerExpressionSyntax.CloseBraceToken.Span;
+                    return !initializerExpressionSyntax.CloseBraceToken.IsMissing;
```

Comment:
i feel like this could be simplified by having a helper function that just returns the close token for these 4 cases.  Then, you can compute `lastDelimitedSpan` and `!...IsMissing` on that value instead of repeating the same code the same way in all 4 cases.

Reply:
I'm OK with the current form (it's already sufficiently simplified from the time when this comment was made).